src/main/scala/lc/core/models.scala

```diff
@@ -6,7 +6,7 @@ case class Parameters(level: String, media: String, input_type: String, size: Op
 case class Id(id: String) extends ChallengeResult
 case class Image(image: Array[Byte]) extends ChallengeResult
 case class Answer(answer: String, id: String)
-case class Result(result: String)
+case class Result(result: String) extends ChallengeResult
```

Comment:
Please rename `Result` to `Success`.

Secondly, `ChallengeResult` has many sub-classes: `Id, Image, Result, Error`

Are all four types possible wherever `ChallengeResult` is expected? I suspect two different use-cases are being conflated here. If so, please create separate types for them.

Reply:
Yeah, in `def getResponse(response: ChallengeResult): Response`. Since it's a generic method, any of the response types are possible.